components/sup/src/manager/service/mod.rs

```
@@ -370,24 +417,12 @@ impl Service {
     /// Run reconfigure hook if present. Return false if it is not present, to trigger default
     /// restart behavior.
     pub fn reconfigure(&mut self) {
-        match self.config.write() {
-            Ok(true) => {
-                self.needs_restart = true;
-                if let Some(err) = self.run_hook(HookType::Reconfigure).err() {
-                    outputln!(preamble self.service_group,
-                              "Reconfiguration hook failed: {}",
-                              err);
-                }
-                if let Some(err) = self.copy_run().err() {
-                    outputln!(preamble self.service_group, "Failed to copy run hook: {}", err);
-                }
-            }
-            Ok(false) => (),
-            Err(e) => {
-                outputln!(preamble self.service_group,
-                          "Failed to write service configuration: {}",
-                          e);
-            }
+        if let Some(err) = self.run_hook(HookType::Reconfigure).err() {
+            outputln!(preamble self.service_group,
+                      "Reconfiguration hook failed: {}",
+                      err);
+        } else {
+            self.needs_reconfiguration = false;
```

@adamhjk I didn't think of a "let it crash" model for the hooks. I think either way will work as long as it's documented